compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -48,6 +51,46 @@ use rustc_hir::intravisit::Visitor;
 use std::cmp::{self, Ordering};
 use std::iter;
 
+#[derive(Debug)]
+struct ConfusablesDirective {
+    names: Vec<String>,
+}
+
+#[derive(Debug)]
+struct ConfusablesDirectiveParseErr {
+    kind: ConfusablesDirectiveParseErrKind,
+    span: Span,
+}
+
+#[derive(Debug)]
+enum ConfusablesDirectiveParseErrKind {
+    InvalidArguments,
```

Comment:
I don't think this needs to be a separate type here unless you add more variants.